sound/soc/sof/topology.c

```diff
@@ -890,9 +905,10 @@ static int sof_parse_tokens(struct snd_soc_component *scomp,
 			    struct snd_soc_tplg_vendor_array *array,
 			    int priv_size)
 {
+	int found = 0;
 	int asize;
 
-	while (priv_size > 0) {
+	while (priv_size > 0 && found < count) {
```

Comment:
@plbossart this comment is in some outdated version, but I guess the code hasn't changed that much.

So priv_size is the byte size of the whole area having the tokens. while loop should run as long all the tokens are found or the area ends. 

Array size is subtracted in all loop rounds from priv_size, so we are returning -EINVAL if priv_size goes negative i.e. not all tokens are found and we run out of area. Otherwise we should return 0 success (means all tokens are found).

in this example  "priv_size > 0 and found==count" means we found all tokens before the end of area.